app/scripts/controllers/network/network.js

```diff
@@ -81,30 +81,14 @@ export default class NetworkController extends EventEmitter {
   constructor({ state = {}, infuraProjectId } = {}) {
     super();
 
-    // create stores
-    this.providerStore = new ObservableStore(
-      state.provider || { ...defaultProviderConfig },
-    );
-    this.previousProviderStore = new ObservableStore(
-      this.providerStore.getState(),
-    );
-    this.networkStore = new ObservableStore('loading');
-    // We need to keep track of a few details about the current network
-    // Ideally we'd merge this.networkStore with this new store, but doing so
-    // will require a decent sized refactor of how we're accessing network
-    // state. Currently this is only used for detecting EIP 1559 support but
-    // can be extended to track other network details.
-    this.networkDetails = new ObservableStore(
-      state.networkDetails || {
+    this.store = new ObservableStore({
+      provder: state.provider || { ...defaultProviderConfig },
+      network: 'loading',
+      networkDetails: state.networkDetails || {
         ...defaultNetworkDetailsState,
       },
-    );
-    this.store = new ComposedStore({
-      provider: this.providerStore,
-      previousProviderStore: this.previousProviderStore,
-      network: this.networkStore,
-      networkDetails: this.networkDetails,
     });
+    this._previousProvider = this.store.getState().provider;
```

Comment:
I will consider breaking this out into a separate PR. It seems reasonably distinct.